File: app/src/test/java/com/melean/taskstimetracker/testUtils/MockGenerator.java
package com.melean.taskstimetracker.testUtils;



import com.melean.taskstimetracker.data.database.RealmObjects.TaskEntityRealmObject;

import org.powermock.api.mockito.PowerMockito;

import io.realm.Realm;

import static org.powermock.api.mockito.PowerMockito.mockStatic;
import static org.powermock.api.mockito.PowerMockito.when;

public class MockGenerator {

    public static Realm getMockedRealm() {
        mockStatic(Realm.class);

        Realm mockedRealm = PowerMockito.mock(Realm.class);

        when(mockedRealm.createObject(TaskEntityRealmObject.class)).thenReturn(new TaskEntityRealmObject());

        when(Realm.getDefaultInstance()).thenReturn(mockedRealm);

        return mockedRealm;
    }
}

File: app/src/test/java/com/melean/taskstimetracker/testUtils/MockGenerator.kt
package com.melean.taskstimetracker.testUtils


import com.melean.taskstimetracker.data.database.RealmObjects.TaskEntityRealmObject

import org.powermock.api.mockito.PowerMockito

import io.realm.Realm

import org.powermock.api.mockito.PowerMockito.mockStatic
import org.powermock.api.mockito.PowerMockito.`when`

object MockGenerator {

    val mockedRealm: Realm
        get() {
            mockStatic(Realm::class.java)

            val mockedRealm = PowerMockito.mock(Realm::class.java)

            `when`(mockedRealm.createObject(TaskEntityRealmObject::class.java)).thenReturn(TaskEntityRealmObject())

            `when`(Realm.getDefaultInstance()).thenReturn(mockedRealm)

            return mockedRealm
        }
}
